### DISCUSSION TITLE
Do I need a special import to use custom hooks in React ?

### DISCUSSION THREAD

**Discussion by coopterrones** [pmndrs/zustand#316] [Q&A] [answered]
I get the following error when trying to `useStore` in my functional component 
<img width="1433" alt="Screen Shot 2021-02-25 at 7 53 37 AM" src="https://user-images.githubusercontent.com/67164959/109171157-b8f9fa00-773e-11eb-84cf-af1923db4db3.png">

my code is as follows:

```store.js``` 
````import React from "react";
import create from "zustand";
import { useQuery } from "@apollo/client";
import { GET_ALL_COMPETITIONS } from "./GraphQL/queries";

export const useStore = create((set) => ({
  competition: null,
  getCurrentCompetition: () =>
    set((state) => ({ competition: queryCompetitions() })),
}));

const queryCompetitions = () => {
  const { error, loading, data } = useQuery(GET_ALL_COMPETITIONS);
  return data;
};
```
```functional component```
```
  const [compeitionFile, setCompetitionFile] = useState(null);

  const compeitionState = () => {
    const file = useStore((state) => state.competition);
  };
```


Please let me know if providing further info helps, thanks

**Comment by olavoparno** [comment, marked as answer]
This is a hook rule being broken, you see, in order to use hooks or custom hooks you must follow the rules of:

- either calling it from a react function component e.g. `function MyReactComponent()` or calling from a custom hook preappended with the word `use`, so you would have to have your `queryCompetitions` renamed to `useQueryCompetitions`.

https://reactjs.org/docs/hooks-rules.html

**Reply by coopterrones** [reply]
Thanks! This is great. I figured I was doing something wrong with nesting. I really appreciate your response!
